101-machine-learning-create/azuredeploy.json

```diff
@@ -2,257 +2,641 @@
+            "keyVaultArmId": "[parameters('cmk_keyvault')]",
+            "keyIdentifier": "[parameters('resource_cmk_uri')]"
           }
-      },
-      "cmk_keyvault":{
-        "type": "string",
-        "defaultValue": "",
-        "metadata": {
-          "description": "Specifies the customer managed keyVault arm id."
-        }
-      },
-      "resource_cmk_uri":{
-        "type": "string",
-        "defaultValue": "",
-        "metadata": {
-          "description": "Specifies if the customer managed keyvault key uri."
+        },
+        "hbiWorkspace": "[parameters('confidential_data')]"
+      }
+    },
+    {
+      "condition": "[not(equals(parameters('privateEndpointType'), 'none'))]",
+      "apiVersion": "2019-04-01",
+      "name": "[concat(parameters('workspaceName'), '-PrivateEndpoint')]",
+      "type": "Microsoft.Network/privateEndpoints",
+      "location": "[if(variables('isLocationValid'), parameters('location'), 'InvalidLocation')]",
+      "dependsOn": [
+        "[resourceId('Microsoft.MachineLearningServices/workspaces', parameters('workspaceName'))]",
+        "[variables('subnet')]"
+      ],
+      "properties": {
+        "privateLinkServiceConnections":"[if(equals(parameters('privateEndpointType'), 'AutoApproval'), variables('defaultPEConnections'), json('null'))]",
+        "manualPrivateLinkServiceConnections": "[if(equals(parameters('privateEndpointType'), 'ManualApproval'), variables('defaultPEConnections'), json('null'))]",
+        "subnet": {
+          "id": "[variables('subnet')]"
         }
       }
-  },
-  "variables": {
-      "tenantId": "[subscription().tenantId]",
-      "storageAccount": "[resourceId(parameters('storageAccountResourceGroupName'), 'Microsoft.Storage/storageAccounts', parameters('storageAccountName'))]",
-      "keyVault": "[resourceId(parameters('keyVaultResourceGroupName'), 'Microsoft.KeyVault/vaults', parameters('keyVaultName'))]",
-      "applicationInsights": "[resourceId(parameters('applicationInsightsResourceGroupName'), 'Microsoft.Insights/components', parameters('applicationInsightsName'))]"
-  },
-  "resources": [
-      {
-          "condition": "[equals(parameters('storageAccountNewOrExisting'), 'new')]",
-          "type": "Microsoft.Storage/storageAccounts",
-          "apiVersion": "2019-04-01",
-          "name": "[parameters('storageAccountName')]",
-          "location": "[parameters('location')]",
-          "sku": {
-              "name": "[parameters('storageAccountType')]"
-          },
-          "kind": "StorageV2",
-          "properties": {
-              "encryption": {
-                  "services": {
-                      "blob": {
-                          "enabled": true
-                      },
-                      "file": {
-                          "enabled": true
-                      }
+    },
+    {
+      "condition": "[equals(parameters('privateEndpointType'), 'AutoApproval')]",
+      "type": "Microsoft.Resources/deployments",
```

Comment:
why is this nested?

Reply:
@bmoore-msft Because it's a series of deployments based on private endpoint and all depends on the PE

Reply:
You don't need nesting for dependencies... is there a problem you ran into?

Reply:
this is not resolved

Reply:
@bmoore-msft I took the dns zone and virtual link out. For the dns records copy, it has to stay there